src/elements/content-sidebar/ActivitySidebar.js

```diff
@@ -1108,26 +1114,61 @@ class ActivitySidebar extends React.PureComponent<Props, State> {
+            FILE_ACTIVITY_TYPE_COMMENT,
+            FILE_ACTIVITY_TYPE_TASK,
+            FILE_ACTIVITY_TYPE_VERSION,
+        ];
+
+        if (status === ACTIVITY_FILTER_OPTION_UNRESOLVED || status === ACTIVITY_FILTER_OPTION_RESOLVED) {
+            filters = [FILE_ACTIVITY_TYPE_ANNOTATION, FILE_ACTIVITY_TYPE_COMMENT, FILE_ACTIVITY_TYPE_VERSION];
+        }
+
+        if (status === FILE_ACTIVITY_TYPE_TASK) {
+            filters = [FILE_ACTIVITY_TYPE_TASK, FILE_ACTIVITY_TYPE_VERSION];
+        }
+
+        return filters;
+    };
+
     getFilteredFeedItems = (): FeedItems | typeof undefined => {
         const { feedItems, feedItemsStatusFilter } = this.state;
-        if (!feedItems || !feedItemsStatusFilter || feedItemsStatusFilter === ACTIVITY_FILTER_OPTION_ALL) {
-            return feedItems;
-        }
-        // Filter is completed on two properties (status and type) because filtering on comments (resolved vs. unresolved)
-        // requires looking at item status to see if it is open or resolved. To filter all tasks, we need to look at the
-        // item type. Item type is also used to keep versions in the feed. Task also has a status but it's status will be
-        // "NOT_STARTED" or "COMPLETED" so it will not conflict with comment's status.
-        return feedItems.filter(item => {
-            return (
-                item.status === feedItemsStatusFilter ||
-                item.type === FEED_ITEM_TYPE_VERSION ||
-                item.type === feedItemsStatusFilter
+        const { api, file } = this.props;
+        const { permissions = {} } = file;
+        const feedAPI = api.getFeedAPI(false);
+        feedAPI.file = file;
+
+        // Clear the feedItems before fetching new data to trigger loading indicator
+        this.setState({ feedItems: undefined });
+
+        // Get the file activity filters
+        const filters = this.getFilters(feedItemsStatusFilter);
+
+        feedAPI.fetchFileActivities(permissions, filters, true).then(items => {
+            const parsedFeedItems = getParsedFileActivitiesResponse(items);
+
+            if (!feedItems || !feedItemsStatusFilter || feedItemsStatusFilter === ACTIVITY_FILTER_OPTION_ALL) {
+                return this.setState({ feedItems: parsedFeedItems });
+            }
+
+            // Filter is completed on two properties (status and type) because filtering on comments (resolved vs. unresolved)
+            // requires looking at item status to see if it is open or resolved. Item type is also used to keep versions in the feed.
+            // Task also has a status but it's status will be "NOT_STARTED" or "COMPLETED" so it will not conflict with comment's status.
+            const filteredActivity = parsedFeedItems.filter(
+                item =>
+                    item.status === feedItemsStatusFilter ||
+                    item.type === FEED_ITEM_TYPE_VERSION ||
+                    item.type === feedItemsStatusFilter,
```

Comment:
I think it makes sense why the filtering here is still included in this PR. However, I think it would be helpful to evaluate what changes would need to be made to the API or code here to be able to pass the feed items directly from the API.

Reply:
Yeah, I believe that the api currently only takes the above params; I think the status is a param in the endpoint at the moment (see [here](https://git.dev.box.net/AppServices/end-user-web/blob/93bf1d03c92b2c64596787d6cc07114ceb12e395/schemas/v2/m0/endpoints/uaa/getActivityForFileId.json#L88)), but it's not attached to anything so at the moment, it doesn't do anything. @bfoxx1906 Do you know more about this endpoint? I think we can change the endpoint if needed, but it's probably out of scope for this PR

Reply:
Yeah, I agree that it is out of scope. I looked into the status param on Friday and saw it wasn't attached.

Reply:
Or @Priyanka-Mudiganti do you know what else we would have to change on the endpoint to get what we need directly from the api?